primitives/runtime/src/offchain/storage_lock.rs

```diff
@@ -173,6 +174,19 @@ impl<B: BlockNumberProvider> Default for BlockAndTimeDeadline<B> {
 	}
 }
 
+impl<B: BlockNumberProvider> fmt::Debug for BlockAndTimeDeadline<B>
+	where <B as BlockNumberProvider>::BlockNumber: fmt::Debug
+{
+	fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
+		write!(
+			f,
+			"BlockAndTimeDeadline(block_number={:?}, time={:?})",
+			self.block_number,
+			self.timestamp,
+		)
```

Comment:
```suggestion
		f.debug_struct("BlockAndTimeDeadline")
			.field("block_number", &self.block_number)
			.field("timestamp", &self.timestamp)
			.finish()
```

Using the `debug_struct` helper gives better results in the case of i.e. pretty-print (`:#?`) formatting requests.